java/src/jmri/jmrit/operations/rollingstock/RollingStock.java

```diff
@@ -849,17 +849,7 @@ public void setRfid(String id) {
             log.debug("Setting IdTag for {} to {}", this, id);
             _rfid = id;
             if (!id.equals(NONE)) {
-                try {
-                    IdTag tag = InstanceManager.getDefault(IdTagManager.class).getIdTag(id);
-                    if (tag != null) {
-                        log.debug("Tag {} found", tag);
-                        setIdTag(tag);
-                    } else {
-                        log.error("Tag {} not found", id);
-                    }
-                } catch (NullPointerException e) {
-                    log.error("Tag manager not found");
-                }
+                IdTag tag = InstanceManager.getDefault(IdTagManager.class).provideIdTag(id);
```

Comment:
`setIdTag(tag);` is missing after this line. Static analysis found this error by noticing that the variable `tag` is not used.